### PR TITLE
feat(ingest): extract lineage between SageMaker jobs and models

### PR DESCRIPTION
**Pull request by kevinhu**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py [hunk 1]
```diff
@@ -248,6 +249,20 @@ class SageMakerJob:
...
 
 
+@dataclass
+class ModelJob:
+    """
+    Intermediate representation of a job's related models. Subsequently used by the SageMaker jobs ingestion framework.
+    """
+
+    job_urn: str
+    job_direction: str  # 'training' or 'downstream'
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py:259]: this should probably be an enum

**Review by hsheth2 — CHANGES_REQUESTED** [review]
minor nits but otherwise LGTM

**Review by hsheth2 — APPROVED** [review]
LGTM!

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**